control/control_test.go

```diff
@@ -395,61 +401,101 @@ func TestLoad(t *testing.T) {
+				})
+				Convey("with loaded plugin type as collector", func() {
+					So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
+				})
+			})
+		})
+
 		// Stop our controller so the plugins are unloaded and cleaned up from the system
 		c.Stop()
 		time.Sleep(100 * time.Millisecond)
 	} else {
-		fmt.Printf("SNAP_PATH not set. Cannot test %s plugin.\n", fixtures.PluginNameMock2)
+		fmt.Printf("SNAP_PATH not set. Cannot test loading plugins.\n")
 	}
 }
 
 func TestLoadWithSignedPlugins(t *testing.T) {
 	if fixtures.SnapPath != "" {
-		Convey("pluginControl.Load should successufully load a signed plugin with trust enabled", t, func() {
+		Convey("pluginControl.Load with trust enabled", t, func() {
 			c := New(getTestConfig())
 			c.pluginTrust = PluginTrustEnabled
-			c.signingManager = &mocksigningManager{signed: true}
-			lpe := newListenToPluginEvent()
-			c.eventManager.RegisterHandler("Control.PluginLoaded", lpe)
+			c.signingManager = &mocksigningManager{}
 			c.Start()
-			time.Sleep(100 * time.Millisecond)
-			_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
-			Convey("so error on loading a signed plugin should be nil", func() {
-				So(err, ShouldBeNil)
+			Convey("Loading a signed plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
+			})
+			Convey("Loading an unsigned plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock1)
+				Convey("Should return an error", func() {
+					So(err, ShouldNotBeNil)
+				})
 			})
 			// Stop our controller to clean up our plugin
 			c.Stop()
-			time.Sleep(100 * time.Millisecond)
 		})
-		Convey("pluginControl.Load should successfully load unsigned plugin when trust level set to warning", t, func() {
+		Convey("pluginControl.Load when trust level set to warning", t, func() {
 			c := New(getTestConfig())
 			c.pluginTrust = PluginTrustWarn
-			c.signingManager = &mocksigningManager{signed: false}
-			lpe := newListenToPluginEvent()
-			c.eventManager.RegisterHandler("Control.PluginLoaded", lpe)
+			c.signingManager = &mocksigningManager{}
 			c.Start()
-			time.Sleep(100 * time.Millisecond)
-			_, err := load(c, fixtures.PluginPathMock2)
-			Convey("so error on loading an unsigned plugin should be nil", func() {
-				So(err, ShouldBeNil)
+			Convey("Loading a signed plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
+			})
+			Convey("Loading an unsigned plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock1)
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
 			})
 			c.Stop()
-			time.Sleep(100 * time.Millisecond)
 		})
-		Convey("pluginControl.Load returns error with trust enabled and signing not validated", t, func() {
+		Convey("pluginControl.Load with trust disabled", t, func() {
```

Comment:
Only diff shows that as a renaming what can be confusing. All changes in this _TestLoadWithSignedPlugins_ needs to be taken into account to catch what have been changed, mostly these three cases were reorganized and another tests cases were added - I will explain that below:

#### Before there was 3 tests cases (see [code](https://github.com/intelsdi-x/snap/blob/master/control/control_test.go#L406)):
1. Convey("pluginControl.Load should successufully load a signed plugin with trust enabled"
2. Convey("pluginControl.Load should successfully load unsigned plugin when trust level set to warning"
3. Convey("pluginControl.Load returns error with trust enabled and signing not validated"


#### After reorganization and adding another tests cases (see [code](https://github.com/IzabellaRaulin/snap/blob/fixed_flaky_tests/control/control_test.go#L437)):
1.Convey("pluginControl.Load with trust enabled"
1.1. Convey("Loading a signed plugin"
1.2. Convey("Loading an unsigned plugin"

2.Convey("pluginControl.Load when trust level set to warning"
2.1. Convey("Loading a signed plugin"
2.2. Convey("Loading an unsigned plugin"

3.Convey("pluginControl.Load with trust disabled
3.1. Convey("Loading a signed plugin"
3.2. Convey("Loading an unsigned plugin"

Please notice, that:
old_1 was moved to new_1.1
old_2 was moved to new 2.2
old_3 was moved to new_1.2